File: ClientApp/src/app/features/reservations/classes/resources/form/reservation-read-resource.ts
import { Guid } from "guid-typescript"
import { PassengerReadResource } from "./passenger-read-resource"

export class ReservationReadResource {

    reservationId: Guid

    date: string
    email: string
    phones: string
    remarks: string
    adults: number
    kids: number
    free: number
    totalPersons: number
    ticketNo: string

    customer: { id: number, description: string }
    destination: { id: number, description: string }
    driver: { id: number, description: string }
    pickupPoint: { id: number, description: string, port: { id: number, description: string } }
    ship: { id: number, description: string }

    user: { id: number, description: string }

    passengers: PassengerReadResource

}

File: BackEnd/Features/Reservations/Resources/Form/ReservationReadResource.cs
using System;
using System.Collections.Generic;
using BlueWaterCruises.Features.PickupPoints;

namespace BlueWaterCruises.Features.Reservations {

    public class ReservationReadResource {

        public string ReservationId { get; set; }

        public string Date { get; set; }
        public int Adults { get; set; }
        public int Kids { get; set; }
        public int Free { get; set; }
        public int TotalPersons { get; set; }
        public string TicketNo { get; set; }
        public string Email { get; set; }
        public string Phones { get; set; }
        public string Remarks { get; set; }

        public SimpleResource Customer { get; set; }
        public SimpleResource Destination { get; set; }
        public PickupPointWithPortDropdownResource PickupPoint { get; set; }
        public SimpleResource Driver { get; set; }
        public SimpleResource Ship { get; set; }
        public SimpleResource Port { get; set; }

        public UserResource User { get; set; }

        public List<PassengerReadResource> Passengers { get; set; }

    }

}